mysql-async/src/test/java/com/github/jasync/sql/db/MySQLSlowConnectionDelegate.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.jasync.sql.db
+
+import com.github.jasync.sql.db.mysql.codec.MySQLHandlerDelegate
+import com.github.jasync.sql.db.mysql.message.server.AuthenticationSwitchRequest
+import com.github.jasync.sql.db.mysql.message.server.HandshakeMessage
+import com.github.jasync.sql.db.mysql.message.server.EOFMessage
+import com.github.jasync.sql.db.mysql.message.server.ErrorMessage
+import com.github.jasync.sql.db.mysql.message.server.OkMessage
+import io.netty.channel.ChannelHandlerContext
+
+class MySQLSlowConnectionDelegate(
+    private val delegate: MySQLHandlerDelegate,
+    private val onOkSlowdownInMillis: Long
+) : MySQLHandlerDelegate {
```

Comment:
kotlin has a delegate feature, maybe you can use that

Reply:
That certainly removed a lot of boilerplate code, thanks!